packages/xsnap/src/object-inspect.js

```diff
@@ -0,0 +1,489 @@
+/* eslint-disable no-empty */
+/* eslint-disable no-nested-ternary,no-use-before-define */
+// Adapted from object-inspect@1.12.0
+/* global WeakRef */
+/*
+MIT License
+
+Copyright (c) 2013 James Halliday
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+const mapSize = Object.getOwnPropertyDescriptor(Map.prototype, 'size').get;
+const mapForEach = Map.prototype.forEach;
+const setSize = Object.getOwnPropertyDescriptor(Set.prototype, 'size').get;
+const setForEach = Set.prototype.forEach;
+const weakMapHas = WeakMap.prototype.has;
+const weakSetHas = WeakSet.prototype.has;
+const weakRefDeref = WeakRef.prototype.deref;
+const booleanValueOf = Boolean.prototype.valueOf;
+const objectToString = Object.prototype.toString;
+const functionToString = Function.prototype.toString;
+const $match = String.prototype.match;
+const $slice = String.prototype.slice;
+const $replace = String.prototype.replace;
+const $toUpperCase = String.prototype.toUpperCase;
+const $test = RegExp.prototype.test;
+const $concat = Array.prototype.concat;
+const $join = Array.prototype.join;
+const bigIntValueOf = BigInt.prototype.valueOf;
+const getOwnPropertySymbols = Object.getOwnPropertySymbols;
+const symToString = Symbol.prototype.toString;
+// ie, `has-tostringtag/shams
+const toStringTag = Symbol.toStringTag;
+const isEnumerable = Object.prototype.propertyIsEnumerable;
+const dateToISOString = Date.prototype.toISOString;
+
+const gPO = Reflect.getPrototypeOf;
+const hasOwn = Object.prototype.hasOwnProperty;
```

Comment:
Why alias all this stuff? I have some vague ideas, but I don't understand well enough to maintain this list. In particular, the `mapSize` and `setSize` contortions.

Reply:
Is this meant to guard against some other vetted shim mutating one of the standard objects? I like defensive, but is that still in the threat model?

Reply:
> Is this meant to guard against some other vetted shim mutating one of the standard objects? I like defensive, but is that still in the threat model?

This is a light refactoring of the original `object-inspect.js` code.  I didn't rewrite it further because I didn't see the need to.